sys/include/suit/v4/suit.h

```diff
@@ -91,74 +101,191 @@ typedef enum {
     SUIT_DIGEST_TYPE_PREIMAGE   = 4     /**< Pre-image digest */
 } suit_v4_digest_type_t;
 
+/**
+ * @brief SUIT component types
+ *
+ * Unofficial list from
+ * [suit-manifest-generator](https://github.com/ARMmbed/suit-manifest-generator)
+ */
 enum {
-    SUIT_COMPONENT_IDENTIFIER   = 1,
-    SUIT_COMPONENT_SIZE         = 2,
-    SUIT_COMPONENT_DIGEST       = 3,
+    SUIT_COMPONENT_IDENTIFIER   = 1,    /**< Identifier component */
+    SUIT_COMPONENT_SIZE         = 2,    /**< Size component */
+    SUIT_COMPONENT_DIGEST       = 3,    /**< Digest component */
 };
 
 /**
  * @brief SUIT v4 component struct
  */
 typedef struct {
-    uint32_t size;
-    CborValue identifier;
-    CborValue url;
-    CborValue digest;
+    uint32_t size;                      /**< Size */
+    CborValue identifier;               /**< Identifier*/
+    CborValue url;                      /**< Url */
+    CborValue digest;                   /**< Digest */
 } suit_v4_component_t;
 
 /**
  * @brief SUIT manifest struct
  */
 typedef struct {
-    cose_sign_t cose;   /**< COSE signature validation struct */
-    const uint8_t *buf; /**< ptr to the buffer of the manifest */
-    size_t len;         /**< length of the manifest */
-    uint32_t validated; /**< bitfield of validated policies */
-    uint32_t state;     /**< bitfield holding state information */
+    cose_sign_t cose;               /**< COSE signature validation struct */
+    const uint8_t *buf;             /**< ptr to the buffer of the manifest */
+    size_t len;                     /**< length of the manifest */
+    uint32_t validated;             /**< bitfield of validated policies */
+    uint32_t state;                 /**< bitfield holding state information */
 
+    /** List of components in the manifest */
     suit_v4_component_t components[SUIT_V4_COMPONENT_MAX];
     //CborValue components[SUIT_V4_COMPONENT_MAX];
-    unsigned components_len;
-    int component_current;
-    riotboot_flashwrite_t *writer;
+    unsigned components_len;        /**< Current number of components */
+    int component_current;          /**< Current component index */
+    riotboot_flashwrite_t *writer;  /**< Pointer to the riotboot flash writer */
+    /** Manifest validation buffer */
     uint8_t validation_buf[SUIT_COSE_BUF_SIZE];
-    cose_key_t *key;    /**< Ptr to the public key for validation */
-    char *urlbuf;
-    size_t urlbuf_len;
+    cose_key_t *key;                /**< Ptr to the public key for validation */
+    char *urlbuf;                   /**< Buffer containing the manifest url */
+    size_t urlbuf_len;              /**< Length of the manifest url */
 } suit_v4_manifest_t;
 
-#define SUIT_MANIFEST_HAVE_COMPONENTS 0x1
-#define SUIT_MANIFEST_HAVE_IMAGE 0x2
+/**
+ * @brief Mask used to determine if SUIT manifest contains components
```

Comment:
(also with the doc below) I think I would prefer to call these bit flags instead of masks